e2e/studio/features/index-advisor.spec.ts

```diff
@@ -0,0 +1,554 @@
+      console.log('Failed to clean up test table:', (e as Error).message)
+    }
+
+    await page.close()
+  })
+
+  test.describe('Check Extension Status', () => {
+    test('should check Index Advisor extensions via Database > Extensions page', async ({
+      ref,
+    }) => {
+      await page.goto(toUrl(`/project/${ref}/database/extensions`))
+
+      try {
+        await waitForApiResponse(page, 'pg-meta', ref, 'extensions')
+      } catch (e) {
+        console.warn('Extensions API timeout, continuing anyway:', (e as Error).message)
+        await page.waitForLoadState('networkidle', { timeout: 10000 }).catch(() => {})
+      }
+
+      await expect(
+        page.getByRole('heading', { name: 'Database Extensions' }),
+        'Database Extensions heading should be visible'
+      ).toBeVisible()
+
+      const searchInput = page.getByPlaceholder('Search for an extension')
+
+      // Search for index_advisor
+      await searchInput.clear()
+      await searchInput.fill('index_advisor')
+
+      const indexAdvisorRow = page.getByRole('row', { name: /index_advisor/i }).first()
+      await expect(indexAdvisorRow, 'index_advisor extension should be in the list').toBeVisible({
+        timeout: 5000,
+      })
+
+      // Search for hypopg
+      await searchInput.clear()
+      await searchInput.fill('hypopg')
+
+      const hypopgRow = page.getByRole('row', { name: /hypopg/i }).first()
+      await expect(hypopgRow, 'hypopg extension should be in the list').toBeVisible({
+        timeout: 5000,
+      })
+
+      // Clear search
+      await searchInput.clear()
+    })
+  })
+
+  test.describe('Enable Index Advisor', () => {
+    test('should display Index Advisor banner when not enabled', async ({ ref }) => {
+      if (isIndexAdvisorEnabled) {
+        test.skip(true, 'Index Advisor is already enabled')
+      }
+
+      // Navigate to query performance page
+      await page.goto(toUrl(`/project/${ref}/observability/query-performance`))
+      await page.waitForLoadState('networkidle')
+
+      await expect(
+        page.getByRole('heading', { name: 'Query Performance' }),
+        'Query Performance heading should be visible'
+      ).toBeVisible()
+
+      // Check for the banner
+      const banner = page.locator('.fixed.bottom-4.right-4')
+
+      // Wait for potential banner to appear or timeout quickly
+      try {
+        await banner.waitFor({ state: 'visible', timeout: 3000 })
+
+        // Hover over the banner to make it interactive
+        await banner.hover()
+
+        // Wait for banner content to be visible after hover
+        const enableText = page.getByText('Enable Index Advisor')
+        await expect(
+          enableText,
+          'Enable Index Advisor text should be visible in banner'
+        ).toBeVisible({ timeout: 2000 })
+      } catch (e) {
+        // Banner not present, which is fine - it may not always show
+        console.log('Banner not visible, which is acceptable')
+      }
+    })
```

Comment:
I think we can remove this test? its just testing the location of the banner which isn't needed IMO